modules/objc/generator/gen_objc.py

```diff
@@ -1544,6 +1542,39 @@ def sanitize_documentation_string(doc, type):
             break
     return "/**\n " + "\n ".join(lines) + "\n */" if hasValues else ""
 
+# copytree from shutil cannot handle an error when directory exists in Python < 3.8
+def _mkdir(newdir):
+    if os.path.isdir(newdir):
+        pass
+    elif os.path.isfile(newdir):
+        raise OSError("A file with the same name as directory "
+                      "'%s' already exists." % dst)
+    else:
+        head, tail = os.path.split(newdir)
+        if head and not os.path.isdir(head):
+            _mkdir(head)
+        if tail:
+            os.mkdir(newdir)
+
+def copytree(src, dst, symlinks=False):
```

Comment:
I believe we could use https://docs.python.org/3/library/shutil.html#shutil.copytree

May be with Python 3.8 if we need `dirs_exist_ok` parameter.

BTW,
`macosx-1` has Python 3.9
`macosx-2` has Python 3.7

Reply:
Yes, but I thought it can work only with python>=3.8, which is not good for versions before 3.8.

Reply:
> copytree cannot write to the directory, if it's already exist

This is why I mention about `dirs_exist_ok` parameter.

I expecting only this change:

```
if sys.version_info >= (3, 8):  # Python 3.8+
    def copy_tree(src, dst):
        shutil.copytree(src, dst, dirs_exist_ok=True)
else:
    from distutils.dir_util import copy_tree
```

Reply:
Is it a good idea to support `distutils` anyway? As far as I know, it will be deprecated in Python 3.12, what is okay to use this (your suggestion) workaround.